libs/designer/src/lib/ui/panel/panelRoot.tsx

```diff
@@ -61,6 +61,13 @@ export const PanelRoot = (props: PanelRootProps): JSX.Element => {
     [currentPanelMode]
   );
 
+  const customPanelLocations: CustomPanelLocation[] = [];
+  for (const customPanelLocation of customPanelLocations) {
+    if (currentPanelMode === customPanelLocation.panelMode) {
+      commonPanelProps.panelLocation = customPanelLocation.panelLocation;
```

Comment:
please move this into a local variable instead of changing the props themselves.

Reply:
ok, putting in memo function in next commit